build/js/live-editor.output.js

```diff
@@ -529,11 +533,13 @@ window.LiveEditorOutput = Backbone.View.extend({
             // Normal case
             } else {
                 // This is debounced (async)
-                this.test(userCode, this.validate, errors, function(errors, testResults) {
-                    this.results.errors = errors;
-                    this.results.tests = testResults;
-                    this.phoneHome();
-                }.bind(this));
+                if (this.validate !== "") {
+                    this.test(userCode, this.validate, errors, function(errors, testResults) {
+                        this.results.errors = errors;
+                        this.results.tests = testResults;
+                        this.phoneHome();
+                    }.bind(this));
```

Comment:
Only test the code using the test worker if ```validate``` is not empty.  This saves cycles when the user is not doing a challenge.